outlines/ui-ux-outline.md

```diff
@@ -0,0 +1,32 @@
+# UI / UX
+
+1. Building reusable components to encapsulate UX patterns
+  1. Pure vs impure in general
```

Comment:
Yeah we've gone back and forth on this.

Smart is bad, because no-one wants to be a "dumb component".

I think "wrapper" is used a bit in the React world these days.

I like pure/impure because it fits pretty well with the functional programming sense of the term. 

We could also use complimentary terms in both cases and call them "smart" and "pure" components?